announcement/announcement-tool/tool/src/java/org/sakaiproject/announcement/tool/AnnouncementAction.java

```diff
@@ -279,26 +282,43 @@ public static String getCurrentOrder() {
 	}
 
 	/**
+	 * Gets a security advisor for reading announcements.
+	 * If <code>announcement.merge.visibility.strict</code> is set to <code>false</code>that allows messages from
+	 * other channels to be read when the current user
+	 * doesn't have permission. This is used to allow messages from merged sites to appear without the
+	 * current user having to be a member.
+	 * @param channelReference The entity reference of the channel in another site.
+	 * @return A security advisor that allows the current user access to that content.
+     */
+	SecurityAdvisor getChannelAdvisor(final String channelReference)
+	{
+		if (ServerConfigurationService.getBoolean("announcement.merge.visibility.strict", false))
```

Comment:
There are already a few calls so I didn't want to change too much of the file, but yes it would be better todo that.